### PR TITLE
Add shared::dxgi1_5

### PR DESCRIPTION
**Pull request by ekicyou**
translate was "shared/dxgi1_5.h".
It is a difference of the dev branch.

Purpose is "um::dcomp" implementation.

### CODE REVIEW

File: src/shared/dxgi1_5.rs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+use um::unknwnbase::IUnknown;
+use um::winnt::HRESULT;
+
+RIDL!(#[uuid(0x80A07424, 0xAB52, 0x42EB, 0x83, 0x3C, 0x0C, 0x42, 0xFD, 0x28, 0x2D, 0x98)]
+interface IDXGIOutput5(IDXGIOutput5Vtbl): IDXGIOutput4(IDXGIOutput4Vtbl) {
+    fn DuplicateOutput1(pDevice: *mut IUnknown,
+                        Flags: UINT,
+                        SupportedFormatsCount: UINT,
+                        pSupportedFormats: *const DXGI_FORMAT,
+                        ppOutputDuplication: *mut *mut IDXGIOutputDuplication
```
Comment (retep998) [inline src/shared/dxgi1_5.rs]: Please use block indentation.

File: src/shared/dxgi1_5.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+RIDL!(#[uuid(0x7632e1f5, 0xee65, 0x4dca, 0x87, 0xfd, 0x84, 0xcd, 0x75, 0xf8, 0x83, 0x8d)]
+interface IDXGIFactory5(IDXGIFactory5Vtbl): IDXGIFactory4(IDXGIFactory4Vtbl) {
+    fn CheckFeatureSupport(
+        Feature: DXGI_FEATURE,
+        pFeatureSupportData: *mut c_void,
+        FeatureSupportDataSize: UINT
+    ) -> HRESULT
+});
+
+DEFINE_GUID!{IID_IDXGIOutput5,0x80A07424,0xAB52,0x42EB,0x83,0x3C,0x0C,0x42,0xFD,0x28,0x2D,0x98}
```
Comment (retep998) [inline src/shared/dxgi1_5.rs]: The formatting of this needs work. For an example of correct formatting see https://github.com/retep998/winapi-rs/blob/dev/src/shared/devguid.rs

**Comment by ekicyou** [comment]
updated.
It is now planned to 

* "shared::dcomptypes"
* "um:dcompanimation"
* "um:dcomp".

Must make a "dcomp-sys" at the end, but not well understood how to create.

**Comment by retep998** [comment]
>Must make a "dcomp-sys" at the end, but not well understood how to create.

Functions are no longer going in -sys crates due to the 0.3 overhaul. Instead you define the functions in the modules directly in winapi like everything else.

**Comment by ekicyou** [comment]
> Functions are no longer going in -sys crates due to the 0.3 overhaul.

Ok. Gather the pieces first.

**Comment by retep998** [comment]
PR has conflicts.

**Comment by ekicyou** [comment]
[dev branch] AppVeyor build failed.
Do wait for build success?
If in a failed state again conflict.

**Comment by retep998** [comment]
The CI failure is my fault. I'll fix it today.

**Comment by ekicyou** [comment]
Clear conflicts. pass msvc target.

**Comment by retep998** [comment]
Could you perhaps *not* run `build.rs` through rustfmt? I really don't like the result.

**Comment by ekicyou** [comment]
Oh, I'm sorry. reproduces.

**Comment by ekicyou** [comment]
repair request, Add shared::dxgi1_5(rev2) #387
#385 close.

**Pull request closed without merging**